homeassistant/components/mqtt/cover.py

```diff
@@ -217,6 +218,8 @@ def __init__(self, hass, config, config_entry, discovery_data):
         self._optimistic = None
         self._tilt_value = None
         self._tilt_optimistic = None
+        self._has_tilt_topic = None
+        self._force_tilt_optimistic = False
```

Comment:
Instead of this, check if `CONF_TILT_STATE_OPTIMISTIC` is set by user or not.

Reply:
Agree

Reply:
You may have to remove the default from `CONF_TILT_STATE_OPTIMISTIC`, unless there's some way to know if the option was set or not (not sure if that's supported by Voluptuous).

Reply:
It defaults to false and thus I see no problem